lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -578,6 +587,93 @@ def partial_update(self, request, comment_id):
         return Response(update_comment(request, comment_id, request.data))
 
 
+class CanPostOrComment(permissions.BasePermission):
+    """
+    Permission that checks to see if the user is allowed to post or
+    comment in the course.
+    """
+
+    def has_permission(self, request, view):
+        """Returns true if the user is enrolled or is staff."""
+        course_key = CourseKey.from_string(view.kwargs.get('course_id'))
+        return (
+            GlobalStaff().has_user(request.user) or
+            CourseStaffRole(course_key).has_user(request.user) or
+            CourseInstructorRole(course_key).has_user(request.user) or
+            CourseEnrollment.is_enrolled(request.user, course_key) or
+            has_discussion_privileges(request.user, course_key)
+        )
```

Comment:
@felipetrz this should requirement / acceptance cretaria of the ticket what permissions to allow so you need to check there and confirm. 

Let me take a look how we can better reuse the existing pieces of code.

Reply:
can I clarify the implication here? you need to be enrolled in the course (or do you need to be instructor, or staff, or global?)

Reply:
edit: I understand now, this list works 💯